Ch14 trad fr ini (#93)
animint/animint-manual-fr: Chapitres/Ch14/Ch14_source.qmd

> # Nommé `clickSelects` et `showSelected`

Author: ```suggestion
# `clickSelects` et `showSelected` nommés
```

<!-- paragraph -->

> Ce chapitre explique comment utiliser [nommé `clickSelects` et `showSelected` variables](/ch06#data-driven-selectors) pour créer des noms de sélecteurs basés sur des données.

Author: ```suggestion
Ce chapitre explique comment utiliser les [variables `clickSelects` et `showSelected` nommées](/ch06#data-driven-selectors) pour créer des noms de sélecteurs basés sur des données.
```

<!-- comment -->
Cette fonctionnalité facilite l'écriture du code animint et accélère la compilation.

<!-- paragraph -->

Plan du chapitre :

<!-- paragraph -->

- Nous commençons par joindre le `PSJ` et en calculant les données à tracer.
<!-- comment -->

> - Nous montrons une méthode pour définir une animint avec de nombreux sélecteurs, en utilisant des boucles for.

Author: ```suggestion
- Nous montrons une méthode pour définir un `animint` avec de nombreux sélecteurs, en utilisant des boucles `for`.
```

> - Nous montrons une méthode pour définir un `animint` avec de nombreux sélecteurs, en utilisant des boucles `for`.

Reviewer: ```suggestion
- Nous présentons une méthode pour définir un `animint` avec de nombreux sélecteurs, en utilisant des boucles `for`.
```

<!-- comment -->

> Cette méthode est techniquement correcte, mais calculatoirement inefficace.

Author: ```suggestion
Cette méthode est techniquement correcte, mais inefficace sur le plan computationnel.
```

> Cette méthode est techniquement correcte, mais inefficace sur le plan computationnel.

Reviewer: https://vitrinelinguistique.oqlf.gouv.qc.ca/fiche-gdt/fiche/8873687/informatique
Le terme computationnel a été utilisé autrefois mais n'est plus en usage.
On pourrait également simplifier la phrase sans perdre d'information essentielle.
```suggestion
Cette approche est techniquement correcte, mais peu efficace.
```

<!-- comment -->

> - Nous expliquons ensuite la méthode à préférer pour définir un `animint` avec de nombreux sélecteurs, en utilisant des `clickSelects` et `showSelected` nommés.

Reviewer: ```suggestion
- Nous expliquons ensuite la méthode à privilégier pour définir un `animint` avec de nombreux sélecteurs, c'est-à-dire en utilisant des `clickSelects` et `showSelected` nommés.
```

<!-- comment -->

> Cette méthode est plus efficace en termes de calcul et plus facile à coder.

Author: ```suggestion
Cette méthode est plus efficace sur le plan computationnel et plus facile à coder.
```

> Cette méthode est plus efficace sur le plan computationnel et plus facile à coder.

Reviewer: ```suggestion
ce qui est plus efficace sur le plan de l'informatique et plus facile à coder.
```
```suggestion
ce qui est plus efficace et plus facile à coder.
```

<!-- paragraph -->

> ## Téléchargement client \<- serveur {#download}

Author: ```suggestion
## Téléchargement du jeu de données {#download}
```

<!-- paragraph -->

> Ci-dessus, nous voyons que `PSJ` est une liste de plusieurs listes et cadres de données.

Author: ```suggestion
Ci-dessus, nous voyons que `PSJ` est une liste de plusieurs listes et tableaux de données.
```

> Ci-dessus, nous voyons que `PSJ` est une liste de plusieurs listes et tableaux de données.

Reviewer: ```suggestion
On peut voir, ci-dessus, que `PSJ` est une liste de plusieurs éléments (listes et tableaux de données).
```

<!-- paragraph -->

> ## Explorer `PSJ` données avec des ggplots statiques {#explore-data-with-static-ggplots}

Author: ```suggestion
## Exploration des données `PSJ` avec des ggplots statiques {#explore-data-with-static-ggplots}
```

<!-- paragraph -->

La figure ci-dessus montre les données brutes et bruitées sous forme d'un `geom_line()` gris.
<!-- comment -->
Les rectangles de couleur représentent des étiquettes qui indiquent si le début ou la fin d'un pic doit être prédit dans une région et un échantillon donnés.
<!-- comment -->
Ensuite, nous ajoutons un panneau pour les problèmes de segmentation, dans lequel un algorithme a cherché un pic commun à tous les échantillons.
<!-- comment -->

> L'algorithme prédit les positions de début et de fin pour un pic de grandes valeurs de données, dans chaque problème.

Reviewer: ```suggestion
Pour chaque problème, l'algorithme prédit les positions de début et de fin du pic de valeur élevée.
```

Reviewer: Je ne suis pas certaine de mon interprétation, cependant placer Pour chaque problème en début de phrase est plus logique en français et valeur élevée est plus juste que grande valeur parce que grande valeur signifie que la donnée est précieuse. Il se peut que ce soit un terme connu de votre domaine par contre...

<!-- comment -->

> Le code ci-dessous calcule un tableau de contingence avec une ligne pour chaque problème de ce type.

Reviewer: ```suggestion
Le code ci-dessous calcule un tableau de contingence avec une ligne pour chaque problème de segmentation.
```

<!-- paragraph -->

> Le code ci-dessus a ajouté la colonne `y` qui est utilisée pour afficher les problèmes dans le code ci-dessous.

Reviewer: ```suggestion
Dans le code ci-dessus une colonne `y` est ajoutée pour l'affichage des problèmes.
```

<!-- paragraph -->

> Ci-dessus, nous voyons un bas de page `problems` a été ajouté au graphique précédent.

Author: ```suggestion
Ci-dessus, on voit qu'un panneau inférieur intitulé `problems` a été ajouté au graphique précédent.
```

<!-- paragraph -->

La figure ci-dessus montre le nombre minimum d'erreurs d'étiquettes en fonction de la taille du problème.
<!-- comment -->

> Les rectangles gris permettent de sélectionner la taille du problème.

Author: ```suggestion
Des rectangles gris seront utilisés pour sélectionner la taille du problème.
```

  log.lambda=mean(log.lambda.range),
  peaks=max(peaks)+0.5)])
(gg.model.selection <- ggplot()+

>    scale_x_continuous("log(penalité)")+

Reviewer: ```suggestion
   scale_x_continuous("log(pénalité)")+
```

<!-- paragraph -->

> La figure ci-dessus montre `errors` (panneau supérieur) et `peaks` (panneau du bas) en fonction de `log(penalty)`.

Author: ```suggestion
La figure ci-dessus montre `errors` (panneau supérieur) et `peaks` (panneau inférieur) en fonction de `log(penalty)`.
```

<!-- comment -->

> Là encore, cette version statique est surchargée ; l'interactivité sera utilisée pour rendre la figure lisible (n'affiche que le sous-ensemble de données correspondant aux valeurs sélectionnées de `bases.per.problem` et `problem.name`).

Reviewer: ```suggestion
Là encore, cette version statique est surchargée ; l'interactivité sera utilisée pour rendre la figure lisible (en n'affichant que le sous-ensemble de données correspondant aux valeurs sélectionnées de `bases.per.problem` et `problem.name`).
```

> Là encore, cette version statique est surreprésentée ; l'interactivité sera utilisée pour que cette figure soit lisible (elle ne montre que le sous-ensemble de données correspondant aux valeurs sélectionnées de`bases.per.problem` et `problem.name`).

Author: ```suggestion
Là encore, cette version statique est surchargée ; l'interactivité sera utilisée pour rendre la figure lisible (n'affiche que le sous-ensemble de données correspondant aux valeurs sélectionnées de `bases.per.problem` et `problem.name`).
```

<!-- paragraph -->

## Visualisation interactive des données (incomplète) {#incomplete}

<!-- paragraph -->

> Dans cette section, nous combinons les `ggplots` de la section précédente en une visualisation des données interactive liée.

Reviewer: Je ne suis pas certaine mais "en" n'est pas clair

> Dans cette section, nous combinons les ggplots de la section précédente en une visualisation des données interactive liée.

Author: ```suggestion
Dans cette section, nous combinons les `ggplots` de la section précédente en une visualisation des données interactive liée.
```

<!-- paragraph -->

La sortie temporelle ci-dessus montre que la définition initiale est rapide.
<!-- comment -->
Générer cette visualisation des données préliminaires (incomplètes) dans le code ci-dessous est également rapide.

<!-- paragraph -->

```{r ch14incomplete}
before.incomplete <- Sys.time()
viz.incomplete
cat(elapsed.incomplete <- Sys.time()-before.incomplete, "secondes\n")
```

<!-- paragraph -->

Nous voyons ci-dessus une visualisation des données avec trois graphiques.

<!-- paragraph -->

- En haut, quatre profils de données ChIP-seq sont affichés, ainsi qu'un panneau de problèmes qui divise l'axe X en problèmes dans lesquels l'algorithme de segmentation s'exécute.
<!-- comment -->
- En cliquant sur le graphique en bas à gauche, on sélectionne la taille du problème, ce qui met à jour les problèmes affichés en haut.
<!-- comment -->
- Le graphique en bas à droite montre le nombre de pics et d'erreurs en fonction de la pénalité (plus grande pour moins de pics).

<!-- paragraph -->

## Ajouter de l'interactivité à l'aide de boucles `for` {#define-using-for-loops}

<!-- paragraph -->

> Dans cette section, nous ajoutons des couches aux `ggplots` précédents en utilisant une boucle `for`, ce qui est sous-optimal, mais nous le montrons à titre de comparaison avec la meilleure approche (`clickSelects` et `showSelected` nommés) qui est présentée dans la prochaine section.

Reviewer: ```suggestion
Dans cette section, nous ajoutons des couches aux `ggplots` précédents en utilisant une boucle `for`, une approche sous-optimale, que nous présentons pour la comparer à une meilleure approche (`clickSelects` et `showSelected` nommés), présentée dans la prochaine section.
```

<!-- comment -->
La visualisation de la section précédente est incomplète.
<!-- comment -->
Nous aimerions ajouter

<!-- paragraph -->

- des rectangles dans le graphique en bas à gauche qui nous permettraient de sélectionner le nombre de pics prédits dans le problème donné.
<!-- comment -->
- des segments et des rectangles dans le graphique du haut qui montreraient les pics prédits et les erreurs d'étiquette.

<!-- paragraph -->

> Une façon (inefficace) d'ajouter ces éléments serait d'utiliser une boucle `for`, codée ci-dessous.

Reviewer: ```suggestion
Une façon (inefficace) d'ajouter ces éléments serait d'utiliser la boucle `for` codée ci-dessous.
```

<!-- comment -->
Pour chaque problème, il existe un sélecteur (appelé `problem.dot`) pour le nombre de pics dans ce problème.
<!-- comment -->

> Dans cette boucle `for`, nous ajoutons donc quelques couches avec `clickSelects="problem.dot"` ou `showSelected="problem.dot"` aux graphiques `coverage` et `modelSelection`.

Reviewer: ```suggestion
Dans cette boucle `for`, nous ajoutons donc quelques couches aux graphiques `coverage` et `modelSelection` à l'aide de `clickSelects="problem.dot"` ou `showSelected="problem.dot"` .
```

<!-- paragraph -->

Nous voyons ci-dessus une visualisation des données avec trois graphiques.

<!-- paragraph -->

- En haut, quatre profils de données ChIP-seq sont affichés, ainsi qu'un panneau de problèmes qui divise l'axe X en problèmes dans lesquels l'algorithme de segmentation s'exécute.
<!-- comment -->
- En cliquant sur le graphique en bas à gauche, on sélectionne la taille du problème, ce qui met à jour les problèmes affichés en haut.
<!-- comment -->
- Le graphique en bas à droite montre le nombre de pics et d'erreurs en fonction de la pénalité (plus grande pour moins de pics).

<!-- paragraph -->

> ## Ajouter de l'interactivité à l'aide de boucles for {#define-using-for-loops}

Author: ```suggestion
## Ajouter de l'interactivité à l'aide de boucles `for` {#define-using-for-loops}
```

<!-- paragraph -->

Dans cette section, nous ajoutons des couches aux ggplots précédents en utilisant une boucle for, ce qui est sous-optimal, mais nous le montrons à titre de comparaison avec la meilleure approche (nommée `clickSelects` et `showSelected`) qui est présentée dans la section suivante.
<!-- comment -->
La visualisation de la section précédente est incomplète.
<!-- comment -->
Nous aimerions ajouter

<!-- paragraph -->

> - des rectangles dans le tracé en bas à gauche qui nous permettraient de sélectionner le nombre de pics prédits dans le problème donné.

Author: ```suggestion
- des rectangles dans le graphique en bas à gauche qui nous permettraient de sélectionner le nombre de pics prédits dans le problème donné.
```

<!-- comment -->
- des segments et des rectangles dans le graphique du haut qui montreraient les pics prédits et les erreurs d'étiquette.

<!-- paragraph -->

> Une façon (inefficace) d'ajouter ces éléments serait d'utiliser une boucle for, codée ci-dessous.

Author: ```suggestion
Une façon (inefficace) d'ajouter ces éléments serait d'utiliser une boucle `for`, codée ci-dessous.
```

<!-- comment -->
Pour chaque problème, il existe un sélecteur (appelé `problem.dot`) pour le nombre de pics dans ce problème.
<!-- comment -->

> Dans cette boucle for, nous ajoutons donc quelques couches avec `clickSelects="problem.dot"` ou `showSelected="problem.dot"` à l'élément `coverage` et `modelSelection` graphiques.

Author: ```suggestion
Dans cette boucle `for`, nous ajoutons donc quelques couches avec `clickSelects="problem.dot"` ou `showSelected="problem.dot"` aux graphiques `coverage` et `modelSelection`.
```

<!-- paragraph -->

> Notez que la compilation prend plusieurs secondes, car il y a beaucoup de geoms (cliquez sur Afficher le tableau de l'état du téléchargement pour les voir tous).

Author: ```suggestion
Notez que la compilation prend plusieurs secondes, car il y a beaucoup de geoms (cliquez sur « Show download status table » pour les voir tous).
```

<!-- comment -->

> Par rapport à la visualisation des données de la section précédente, celle-ci a

Reviewer: ```suggestion
Par rapport à la visualisation des données de la section précédente, celle-ci présente
```

<!-- paragraph -->

- des segments bleus qui apparaissent dans le graphique des données de couverture supérieure, pour indiquer les pics prédits.
<!-- comment -->
- des rectangles de sélection qui peuvent être cliqués dans le graphique en bas à droite, pour modifier le nombre d'échantillons présentant un pic dans le problème sélectionné.

<!-- paragraph -->

Dans la prochaine section, nous allons créer la même visualisation des données, mais de manière plus efficace.

<!-- paragraph -->

> ## Ajouter de l'interactivité à l'aide de `clickSelects` et `showSelected` nommés {#define-using-named}

Reviewer: Tous les autres sous-titres débutent par des substantifs, pas des verbes à l'infinitif
```suggestion
## Définition de l'interactivité à l'aide de `clickSelects` et `showSelected` nommés {#define-using-named}
```

> ## Ajouter de l'interactivité à l'aide de named `clickSelects` et `showSelected` {#define-using-named}

Author: ```suggestion
## Ajouter de l'interactivité à l'aide de `clickSelects` et `showSelected` nommés {#define-using-named}
```

<!-- paragraph -->

> La sortie ci-dessus montre un tableau de contingence avec une ligne par pic qui peut être affiché, pour différents échantillons, problèmes, et choix interactifs de. `bases.per.problem` et `peaks` de manière interactive.

Author: ```suggestion
La sortie ci-dessus montre un tableau de contingence avec une ligne par pic qui peut être affiché, pour différents échantillons, problèmes, et choix interactifs des paramètres `bases.per.problem` et `peaks`.
```

<!-- comment -->

> Notez que le `problem.dot` qui définit le nom du sélecteur qui stockera le nombre de pics actuellement sélectionné pour ce problème.

Author: ```suggestion
Notez que la colonne `problem.dot` qui définit le nom du sélecteur qui stockera le nombre de pics actuellement sélectionné pour ce problème.
```

<!-- paragraph -->

Dans le code ci-dessous, l'idée principale est que pour chaque problème, il existe un sélecteur défini par la colonne `problem.dot` pour le nombre de pics dans ce problème.
<!-- comment -->
Nous utilisons `showSelected=c(problem.dot="peaks")` et `clickSelects=c(problem.dot="peaks")` pour indiquer que le nom du sélecteur se trouve dans le fichier `problem.dot` et que la valeur de la sélection se trouve dans la colonne `peaks`colonne.
<!-- comment -->

> Le`animint2dir()` crée une variable de sélection pour chaque valeur unique de `problem.dot` (et il utilise les valeurs correspondantes dans `peaks` pour définir/mettre à jour la/les valeur(s) sélectionnée(s)).

Author: ```suggestion
Le compilateur `animint2dir()` crée une variable de sélection pour chaque valeur unique de `problem.dot` (et il utilise les valeurs correspondantes dans `peaks` pour définir/mettre à jour la/les valeur/geoms sélectionnée(s)).
```

<!-- paragraph -->

> La sortie ci-dessus montre un tableau de contingence avec une ligne par pic qui peut être affiché, pour différents échantillons, problèmes, et choix interactifs des paramètres `bases.per.problem` et `peaks`.

Reviewer: ```suggestion
La sortie ci-dessus génère un tableau de contingence qui comporte une ligne par pic affichable, et ce pour différents échantillons, problèmes ou paramètres `bases.per.problem` et `peaks`.
```

<!-- comment -->
Notez que la colonne `problem.dot` qui définit le nom du sélecteur qui stockera le nombre de pics actuellement sélectionné pour ce problème.

<!-- paragraph -->

> Dans le code ci-dessous, l'idée principale est que pour chaque problème, il existe un sélecteur défini par la colonne `problem.dot` pour le nombre de pics dans ce problème.

Reviewer: ```suggestion
L'idée principale du code ci-dessous est celle-ci: pour chaque problème, un sélecteur est défini par la colonne `problem.dot` pour le nombre de pics dans ce problème.
```

<!-- paragraph -->

> Il est clair qu'il faut beaucoup moins de temps pour évaluer le code R ci-dessus qui utilise les `clickSelects` et `showSelected` nommés.

Reviewer: je m'interroge sur évaluer parce que plus bas vous utilisez "le temps d'exécution".

<!-- paragraph -->

> Le tableau de contingence ci-dessus montre que la visualisation des données définie à l'aide de boucles for prend environ deux fois plus d'espace disque que la visualisation des données utilisant des boucles named. `clickSelects` et `showSelected`.

Author: ```suggestion
Le tableau de contingence ci-dessus montre que la visualisation des données définie à l'aide de boucles `for` prend environ deux fois plus d'espace disque que la visualisation des données utilisant des `clickSelects` et `showSelected` nommés.
```

<!-- paragraph -->

Il ressort clairement du tableau de contingence ci-dessus que les `clickSelects` et `showSelected` nommés sont plus efficaces à ces deux égards et devraient être utilisés à la place des boucles `for`.

<!-- paragraph -->

Exercices :

<!-- paragraph -->

- Utilisez les `clickSelects` et `showSelected` nommés pour créer une visualisation illustrant le sur- et sous-ajustement, comme dans [cette visualisation du modèle linéaire et des plus proches voisins](https://tdhock.github.io/2023-12-04-degree-neighbors/).
<!-- comment -->

> - Utilisez les `clickSelects` et `showSelected` nommés pour créer une visualisation de données de votre domaine d'expertise.

Reviewer: ```suggestion
- Utilisez des `clickSelects` et des `showSelected` nommés pour créer une visualisation de données dans votre domaine d'expertise.
```

<!-- paragraph -->

> Suivant, [Chapitre 15](/ch15) explique comment visualiser les algorithmes de recherche de racines.

Author: ```suggestion
Dans le [chapitre 15](/ch15), nous vous expliquerons comment visualiser les algorithmes de recherche de racines (« root-finding ».
```
